packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -632,21 +629,6 @@ async function handler(
         },
       });
     }
-  } else if (queuedFormResponse) {
-    if (routingFormResponseId === undefined) {
-      throw new HttpError({ statusCode: 400, message: "Missing routingFormResponseId" });
-    }
-
-    // If we're not returning the response then we still need
-    // to update the routing form response id with the actual on
-    routingFormResponse = await RoutingFormResponseRepository.writeQueuedFormResponseToFormResponse(
-      routingFormResponseId,
-      bookerEmail
-    );
-    if (!routingFormResponse) {
-      throw new HttpError({ statusCode: 400, message: "Missing routingFormResponse" });
-    }
```

Comment:
Simpified the code with lesser branches and also removed 'Missing routingFormResponse' error which is valid in case of Dry Run.